diff --git a/src/math/src/poly.findroots.c b/src/math/src/poly.findroots.c
index 13c6040..3a3841a 100644
--- a/src/math/src/poly.findroots.c
+++ b/src/math/src/poly.findroots.c
@@ -284,7 +284,7 @@ void POLY(_findroots_bairstow_recursion)(T * _p,
 #endif
 
         // adjust u, v
-        if (isnan(du) || isnan(dv)) {
+        if (isnan((double)du) || isnan((double)dv)) {
             u *= 0.5f;
             v *= 0.5f;
         } else {
