--- host/examples/rfnoc_rx_to_file.cpp	2017-02-24 18:27:49.315919466 -0500
+++ host/examples-patched/rfnoc_rx_to_file.cpp	2017-02-24 18:52:41.358560191 -0500
@@ -389,9 +389,9 @@
             blk_ctrl->set_args(uhd::device_addr_t(block_args));
         }
         // Connect:
-        std::cout << "Connecting " << radio_ctrl_id << " ==> " << blk_ctrl->get_block_id() << std::endl;
-        rx_graph->connect(radio_ctrl_id, radio_chan, blk_ctrl->get_block_id(), uhd::rfnoc::ANY_PORT);
-        streamer_args["block_id"] = blk_ctrl->get_block_id().to_string();
+        std::cout << "Connecting " << radio_ctrl_id << " ==> " << blk_ctrl->unique_id() << std::endl;
+        rx_graph->connect(radio_ctrl_id, radio_chan, blk_ctrl->unique_id(), uhd::rfnoc::ANY_PORT);
+        streamer_args["block_id"] = blk_ctrl->unique_id();
 
         spp = blk_ctrl->get_args().cast<size_t>("spp", spp);
     }

