--- cpp_armv7l/Makefile.am	2016-09-28 16:43:41.500841615 -0400
+++ cpp_armv7l-patched/Makefile.am	2016-10-24 12:42:25.994775773 -0400
@@ -4,7 +4,7 @@
 
 xmldir = $(prefix)/dev/devices/RFNoC_DefaultPersona/
 dist_xml_DATA = ../RFNoC_DefaultPersona.scd.xml ../RFNoC_DefaultPersona.prf.xml ../RFNoC_DefaultPersona.spd.xml
-ACLOCAL_AMFLAGS = -I m4 -I${OSSIEHOME}/share/aclocal/ossie
+ACLOCAL_AMFLAGS = -I m4
 AUTOMAKE_OPTIONS = subdir-objects
 
 distclean-local:
